Update versioning-policy.md to add the type addition condition (#38)
meilisearch/engine-team: resources/versioning-policy.md

🚨 Any change about the relevancy that is related to API usage, and thus, that may impact users to change their code (for instance changing the default `matchingStrategy` value) is not related to this specific section and would lead us to increase the MAJOR.

### New "variant" type addition

We don't consider breaking to add a new type to an already existing list of variant. For example, adding a new type of `task`, or a new type of error `code`.

> We are aware some strongly typed language code bases could be impacted, and our recommendation is to set an `unknown` type when possible.

Reviewer: ```suggestion
We are aware some strongly typed language code bases could be impacted, and our recommendation is to handle the possibility of an unknown type when deserializing the answers of meilisearch.
```

I feel like your sentence is giving a solution on how to handle this instead of just letting people know it will happen :thinking: 
I don't like the formulation of mine either, but I feel like it's easier to understand where the issue will arise, and then it lets you conceive your own solution how you feel it should work with your language (for example, in rust it won't be a type but simply an enum variant, while in C# it's probably going to be a class?)

tldr: I don't like the word `set` and the fact that `unknown` is in a code block :joy:

Author: ```suggestion
We are aware some strongly typed language code bases could be impacted, and our recommendation is to handle the possibility of an unknown type when deserializing Meilisearch's response.
```

Author: I like your suggestion, I just made some small changes 👌

> We are aware some strongly type language code base could be impacted, and we always would recommend setting an `unknown` type when working with strongly typed language and Meilisearch.

Reviewer 2: A suggestion about the strongly typed stacks...
```suggestion
We are aware some strongly typed language code bases could be impacted, and our recommendation is to set an `unknown` type when possible.
```

> We are aware some strongly typed language code bases could be impacted, and our recommendation is to handle the possibility of an unknown type when deserializing Meilisearch's response.

Reviewer 2: What about this?
```suggestion
We are aware some strongly typed language code bases could be impacted, and our recommendation is to handle the possibility of having an unknown type when deserializing Meilisearch's response.
```

I got @irevoire's idea, and I agree with it. I think adding this verb will tell the user that they can expect an unknown type and they should handle it by themselves.
